kerastuner/tuners/bayesian.py

```diff
@@ -0,0 +1,289 @@
+            if hp.name == name:
+                return index
+        return None
+
+    def _upper_confidence_bound(self, x):
+        x = x.reshape(-1, self._dim)
+        try:
+            mu, sigma = self.gpr.predict(x, return_std=True)
+        except UserWarning:
+            mu = np.array([1.0] * len(x), dtype=np.float64)
+            sigma = np.array([0.1] * len(x), dtype=np.float64)
+        return mu - self.beta * sigma
+
+    @property
+    def _dim(self):
+        return len([hp for hp in self.space if not isinstance(hp, hp_module.Fixed)])
+
+    def _generate_vector(self, ):
+        min_val = float('inf')
+        min_x = None
+        bounds = []
+        for hp in self.space:
+            if isinstance(hp, hp_module.Choice):
+                bound = [0, len(hp.values)]
+            elif isinstance(hp, hp_module.Fixed):
+                # Fixed values are excluded form the vector.
+                continue
+            else:
+                bound = [hp.min_value, hp.max_value]
+            bounds.append(bound)
+        bounds = np.array(bounds)
+
+        # Find the best optimum by starting from n_restart different random points.
+        n_restarts = 25
+        for x in np.random.uniform(bounds[:, 0], bounds[:, 1],
+                                   size=(n_restarts, self._dim)):
+            res = scipy_optimize.minimize(self._upper_confidence_bound,
+                                          x0=x, bounds=bounds,
+                                          method='L-BFGS-B')
+            if res.fun < min_val:
+                min_val = res.fun[0]
+                min_x = res.x
+
+        return min_x.reshape(-1, 1)
+
+
+class BayesianOptimization(tuner_module.Tuner):
+    """BayesianOptimization tuning with Gaussian process.
+
+    Args:
+        hypermodel: Instance of HyperModel class
+            (or callable that takes hyperparameters
+            and returns a Model instance).
+        objective: String. Name of model metric to minimize
+            or maximize, e.g. "val_accuracy".
+        max_trials: Int. Total number of trials
+            (model configurations) to test at most.
+            Note that the oracle may interrupt the search
+            before `max_trial` models have been tested.
+        init_samples: Int. The number of randomly generated samples as initial
+            training data for Bayesian optimization.
+        alpha: Float or array-like. Value added to the diagonal of
+            the kernel matrix during fitting.
+        beta: Float. The balancing factor of exploration and exploitation.
+            The larger it is, the more explorative it is.
+        seed: Int. Random seed.
+    """
+
+    def __init__(self,
+                 hypermodel,
+                 objective,
+                 max_trials,
+                 init_samples=2,
+                 seed=None,
+                 **kwargs):
+        oracle = BayesianOptimizationOracle(num_initial_points=init_samples,
```

Comment:
This argument should have the same name across the tuner and the oracle (init_samples)